api/v1alpha1/ndb_api_helpers.go

```diff
@@ -75,160 +75,124 @@ func GenerateProvisioningRequest(ctx context.Context, ndbclient *ndbclient.NDBCl
-			DatabaseType:             GetDatabaseEngineName(dbSpec.Instance.Type),
-			Name:                     dbSpec.Instance.DatabaseInstanceName,
-			DatabaseDescription:      "Database provisioned by ndb-operator: " + dbSpec.Instance.DatabaseInstanceName,
-			SoftwareProfileId:        profilesMap[PROFILE_TYPE_SOFTWARE].Id,
-			SoftwareProfileVersionId: profilesMap[PROFILE_TYPE_SOFTWARE].LatestVersionId,
-			ComputeProfileId:         profilesMap[PROFILE_TYPE_COMPUTE].Id,
-			NetworkProfileId:         profilesMap[PROFILE_TYPE_NETWORK].Id,
-			DbParameterProfileId:     profilesMap[PROFILE_TYPE_DATABASE_PARAMETER].Id,
-			NewDbServerTimeZone:      dbSpec.Instance.TimeZone,
-			CreateDbServer:           true,
-			NodeCount:                1,
-			NxClusterId:              dbSpec.NDB.ClusterId,
-			SSHPublicKey:             SSHPublicKey,
-			Clustered:                false,
-			AutoTuneStagingDrive:     true,
-			TimeMachineInfo: TimeMachineInfo{
-				Name:             dbSpec.Instance.DatabaseInstanceName + "_TM",
-				Description:      sla.Description,
-				SlaId:            sla.Id,
-				Schedule:         make(map[string]string),
-				Tags:             make([]string, 0),
-				AutoTuneLogDrive: true,
+			{
+				Name:  "database_names",
+				Value: database_names,
 			},
-			ActionArguments: []ActionArgument{
-				{
-					Name:  "proxy_read_port",
-					Value: "5001",
-				},
-				{
-					Name:  "listener_port",
-					Value: "5432",
-				},
-				{
-					Name:  "proxy_write_port",
-					Value: "5000",
-				},
-				{
-					Name:  "database_size",
-					Value: strconv.Itoa(dbSpec.Instance.Size),
-				},
-				{
-					Name:  "auto_tune_staging_drive",
-					Value: "true",
-				},
-				{
-					Name:  "enable_synchronous_mode",
-					Value: "false",
-				},
-				{
-					Name:  "backup_policy",
-					Value: "primary_only",
-				},
-				{
-					Name:  "dbserver_description",
-					Value: "dbserver for " + dbSpec.Instance.DatabaseInstanceName,
-				},
-				{
-					Name:  "database_names",
-					Value: database_names,
-				},
-				{
-					Name:  "db_password",
-					Value: dbPassword,
-				},
+			{
+				Name:  "db_password",
+				Value: dbPassword,
 			},
-			Nodes: []Node{
-				{
-					Properties: make([]string, 0),
-					VmName:     dbSpec.Instance.DatabaseInstanceName + "_VM",
-				},
+		},
+	}
+	// Setting action arguments based on database type
+	dbActionArgsMap := map[string][]ActionArgument{
+		DATABASE_TYPE_MONGODB: {
+			{
+				Name:  "listener_port",
+				Value: "27017",
 			},
-		}
+			{
+				Name:  "database_size",
+				Value: strconv.Itoa(dbSpec.Instance.Size),
+			},
+			{
+				Name:  "log_size",
+				Value: "100",
+			},
+			{
+				Name:  "journal_size",
+				Value: "100",
+			},
+			{
+				Name:  "restart_mongod",
+				Value: "true",
+			},
+			{
+				Name:  "working_dir",
+				Value: "/tmp",
+			},
+			{
+				Name:  "db_user",
+				Value: dbSpec.Instance.DatabaseInstanceName,
+			},
+		},
+		DATABASE_TYPE_POSTGRES: {
```

Comment:
Please try to avoid using a map for hardcoding the values. Thinking about extensibility, in future we might need to provide the values of some of these action arguments based on a value from the user input then it would be tedious to handle that in this design choice. Perhaps think of a factory model?